### PR TITLE
Support Spark 4.0

### PR DESCRIPTION
**Pull request by pawel-big-lebowski**
Support Spark 4.0 preview -> https://spark.apache.org/news/spark-4.0.0-preview1.html

### CODE REVIEW

File: integration/spark/app/src/main/java/io/openlineage/spark/agent/OpenLineageSparkListener.java [hunk 1]
```diff
@@ -174,7 +175,7 @@ public void onJobStart(SparkListenerJobStart jobStart) {
             .map(Integer.class::cast)
             .collect(Collectors.toSet());
 
-    if (sparkVersion.startsWith("3")) {
+    if (SPARK_VERSION_3.compareTo(sparkVersion) < 0) {
```
Comment (mobuchowski) [inline integration/spark/app/src/main/java/io/openlineage/spark/agent/OpenLineageSparkListener.java:178]: Is this really more clear? Does not seem to me...

File: integration/spark/app/src/test/java/io/openlineage/spark/agent/GoogleCloudIntegrationTest.java [hunk 2]
```diff
@@ -56,8 +58,8 @@ class GoogleCloudIntegrationTest {
   private static final String NAMESPACE = "google-cloud-namespace";
   private static final int MOCKSERVER_PORT = 3000;
   private static final String LOCAL_IP = "127.0.0.1";
-  private static final String SPARK_3 = "(3.*)";
-  private static final String SPARK_3_3 = "(3\\.[3-9].*)";
+  private static final String GREATER_THAN_SPARK2 = "(3.*)";
+  private static final String GREATER_THAN_SPARK3_3 = "(3\\.[3-9].*)";
```
Comment (mobuchowski) [inline integration/spark/app/src/test/java/io/openlineage/spark/agent/GoogleCloudIntegrationTest.java:62]: Should those both contain Spark 4? If not, why?

**Review by mobuchowski — COMMENTED** [review]
Few things to look at. Besides looking good.

File: integration/spark/app/src/test/java/io/openlineage/spark/agent/GoogleCloudIntegrationTest.java (hunk 2 above)
Comment (pawel-big-lebowski) [reply integration/spark/app/src/test/java/io/openlineage/spark/agent/GoogleCloudIntegrationTest.java:62]: As mentioned few lines above:
```
// TODO: Please note the test remains disabled for Spark 4.0 for now (no applicable connector
// version available)
```

File: integration/spark/app/src/main/java/io/openlineage/spark/agent/OpenLineageSparkListener.java (hunk 1 above)
Comment (pawel-big-lebowski) [reply integration/spark/app/src/main/java/io/openlineage/spark/agent/OpenLineageSparkListener.java]: moving this to `SparkVersionsUtils`

**Review by mobuchowski — APPROVED** [review]

**Pull request merged**